### DISCUSSION TITLE
Problem with big custom creatures that can enter sub

### DISCUSSION THREAD

**Discussion by SomeRandomNoobKekeke** [FakeFishGames/Barotrauma#12680] [Code questions and modding support]
When big creature, what allowed to board subs, touches any hull with any limb, it's considered inside. 
It stop colliding with level walls, and can fall through and get stuck

[poor girls, they just wanted a bite](https://www.youtube.com/watch?v=KQGLl89sbAo)

This doesn't happen in a normal game because big creatures are not allowed in sub, and all that allowed are smaller than smallest wall cell and can't get stuck

But it's limiting ability to make big scary creatures that can eat you from sub

What's your ideas on how to solve this?

e.g. add some option to creature prefab like "alwaysCollide", so modders could set it for their abominations

**Comment by Hazmat616** [comment]
We need hard and soft walls. So the outer hull is soft and won't prevent anything from entering after it's broken. But inner floors and walls will stop big things. This way they can sometimes get you. Alternatively give hull and walls two health bars with the 2nd being damaged only by creatures that can't enter the sub and when it's broken completely it won't stop them. This way it can be independently balanced since the first health bar stays the same and lets small things in the 2nd can be whatever is deemed appropriate for large things.
